src/test/java/org/altbeacon/beacon/distance/ModelSpecificDistanceCalculatorTest.java

```diff
@@ -58,4 +52,60 @@ public void testSelectsNexus4OnExactMatch() {
         assertEquals("should be Nexus 4", "Nexus 4", distanceCalculator.getModel().getModel());
     }
 
+	@Test
+	public void testConcurrentModificationException() {
+		org.robolectric.shadows.ShadowLog.stream = System.err;
+
+		final Context applicationContext = ShadowApplication.getInstance().getApplicationContext();
+
+		final AndroidModel model = new AndroidModel("4.4.2", "KOT49H", "Nexus 4", "LGE");
+		final ModelSpecificDistanceCalculator distanceCalculator =
+				new ModelSpecificDistanceCalculator(applicationContext, null, model);
+
+		Runnable runnable2 = new Runnable() {
+			@Override
+			public void run() {
+				try {
+					while (true) {
+						distanceCalculator._buildModelMapTest("{\n" +
+															  "  \"models\":\n" +
+															  "  [\n" +
```

Comment:
Here is sample JSON from test package :)

{\n" +
+															  "      \"coefficient1\": 0.42093,\n" +
+															  "      \"coefficient2\": 6.9476,\n" +
+															  "      \"coefficient3\": 0.54992,\n" +
+															  "      \"version\":\"4.4.2\",\n" +
+															  "      \"build_number\":\"KOT49H\",\n" +
+															  "      \"model\":\"Nexus 4\",\n" +
+															  "      \"manufacturer\":\"LGE\"\n" +
+															  "    },\n" +
+															  "    {\n" +
+															  "      \"coefficient1\": 0.42093,\n" +
+															  "      \"coefficient2\": 6.9476,\n" +
+															  "      \"coefficient3\": 0.54992,\n" +
+															  "      \"version\":\"4.4.2\",\n" +
+															  "      \"build_number\":\"LPV79\",\n" +
+															  "      \"model\":\"Nexus 5\",\n" +
+															  "      \"manufacturer\":\"LGE\",\n" +
+															  "      \"default\": true\n" +
+															  "    }\n" +
+															  "  ]\n" +
+															  "}");